src/lib/libast/vmalloc/malloc.c

```diff
@@ -600,7 +600,11 @@ void _vmoptions(int boot)
 
 #if _malloc_hook
 
+#if __FreeBSD__
```

Comment:
Note that macOS also puts some things needed here in *stdlib.h*. It also has a *malloc.h* that defines things like `struct mstats` but it's at *malloc/malloc.h*. I don't think we want to do it this way. There should be a meson test for these two headers. Although I think it is safe to just assume *stdlib.h* is available and unconditionally include it. We probably only need a feature test for the *malloc.h* header.

Reply:
Test for headers will not work because FreeBSD has this definition for `malloc.h` :

```
#error "<malloc.h> has been replaced by <stdlib.h>"
```

That's why I have put a separate check for FreeBSD there.